receiver/snmpreceiver/otel_metric_helper.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snmpreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/snmpreceiver"
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+// generalResourceKey is the resource key for the no general "no attribute" resource
+const generalResourceKey = ""
+
+// getResourceKey returns a unique key based on all of the relevant resource attributes names
+// as well as the current row index for a metric data point (a row index corresponds to values
+// for the attributes that all belong to a single resource, so we only need this for uniqueness
+// and not each value for each attribute name)
```

Comment:
Please add the detailed explanation from your comment on the other PR as a comment.

Reply:
@djaglowski done